src/vs/workbench/contrib/terminalContrib/stickyScroll/browser/terminalStickyScrollOverlay.ts

```diff
@@ -171,25 +171,26 @@ export class TerminalStickyScrollOverlay extends Disposable {
 		if (command.promptStartMarker) {
 			promptStartLine = Math.min(command.promptStartMarker?.line ?? marker.line, marker.line);
 			// Trim any leading whitespace-only lines to retain vertical space
-			while (promptStartLine < marker.line && (this._xterm.raw.buffer.active.getLine(promptStartLine)?.translateToString(true) ?? '').length === 0) {
+			while (promptStartLine < marker.line && (buffer.getLine(promptStartLine)?.translateToString(true) ?? '').length === 0) {
 				promptStartLine++;
 			}
 			promptRowCount = marker.line - promptStartLine + 1;
 		}
 
 		// Clear attrs, reset cursor position, clear right
-		// TODO: Serializing all content up to the required line is inefficient; support providing single line/range serialize addon
-		const s = this._serializeAddon.serialize({
-			scrollback: this._xterm.raw.buffer.active.baseY - promptStartLine
+		const content = this._serializeAddon.serialize({
+			range: {
+				start: promptStartLine,
+				end: promptStartLine + promptRowCount - 1
+			}
 		});
 
 		// Write content if it differs
-		const content = s ? s.substring(0, findNthOccurrenceIndex(s, '\r', promptRowCount)) : undefined;
 		if (content && this._currentContent !== content) {
 			if (this._stickyScrollOverlay.rows !== promptRowCount) {
 				this._stickyScrollOverlay.resize(this._stickyScrollOverlay.cols, promptRowCount);
 			}
-			this._stickyScrollOverlay.write('\x1b[0m\x1b[H\x1b[2K');
+			this._stickyScrollOverlay.write('\x1b[0m\x1b[H\x1b[2J');
```

Comment:
what does this J do?

Reply:
K is clear line, J is clear display (all lines)